zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/NativeUserTaskFormTest.java

```diff
@@ -89,7 +172,32 @@ public void shouldResolveAnIncidentIfFormIsDeployed() {
             tuple(incidentCreated.getKey(), IncidentIntent.CREATED),
             tuple(incidentCreated.getKey(), IncidentIntent.RESOLVED));
 
-    assertUserTaskActivation(processInstanceKey);
+    assertUserTaskActivation(processInstanceKey, form.getFormKey());
+  }
+
+  @Test
+  public void shouldRaiseAnIncidentIfFormIsNotDeployedInSameDeploymentForBindingTypeDeployment() {
+    // given
+    deployForm(TEST_FORM_1);
+    final BpmnModelInstance process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .userTask("task")
+            .zeebeFormId(FORM_ID_1)
```

Comment:
🔧 Same as in other linked resource cases where an incident is thrown for not finding the resource with deployment binding, let's use an expression here to allow adding a validation later.